pycaiso/oasis.py

```diff
@@ -425,3 +425,96 @@ def get_lmps(
         sort_values=["OPR_DT", "OPR_HR"],
         reindex_columns=COLUMNS,
     )
+
+
+
+# Registry of known group reports – 
+_GROUP_REPORTS: dict[str, dict[str, Any]] = {
+    "DAM_LMP_GRP": {"endpoint": "GroupZip", "version": 12, "resultformat": 6},
+}
+
+def _build_groupzip_url(
+    group_id: str, 
+    start: datetime,
+    *,
+    local_tz: str = "America/Los_Angeles",
+    **extra: Any,
+) -> str:
+    """
+    Construct the full GroupZip URL for *group_id* and *trading day*. 
+    Only the daily Day-Ahead LMP group report (**DAM_LMP_GRP**) 
+    is exposed for now, but future additions could include: 
+    RTM_LMP_GRP, RTM_LAP_GRP, LMP_GHG_PRC, PRC_RTM_LAP
+
+    Args—
+    + group_id (str): groupid for fetching a given group report (definitions documented [here](https://www.caiso.com/documents/oasis-interfacespecification_v5_1_2clean_fall2017release.pdf))
+    + start (datetime): Trading day to fetch data for (group reports are always for a single day)
+      
+    Returns—
+    + str: URL to fetch the group report from
+    
+    See official documentation of OASIS API Specs for more details on group reports:
+    https://www.caiso.com/documents/oasis-interfacespecification_v5_1_2clean_fall2017release.pdf
+    """
+    if group_id not in _GROUP_REPORTS:
+        raise ValueError(f"Unknown group_id: {group_id}")
+    
+    spec = _GROUP_REPORTS["DAM_LMP_GRP"]
+    params = {
+        "groupid": group_id,
+        "version": spec["version"],
+        "resultformat": spec["resultformat"],
+        "startdatetime": Oasis()._get_UTC_string(start, local_tz),
+        **extra,
+    }
+    return (
+        f"http://oasis.caiso.com/oasisapi/{spec['endpoint']}?"
+        + "&".join(f"{k}={v}" for k, v in params.items())
+    )
+    
+
+def fetch_groupzip(
+    group_id: str,
+    trading_day: datetime,
+    *,
+    local_tz: str = "America/Los_Angeles",
+) -> pd.DataFrame:
+    """
+    Download all CSVs in the GroupZip bundle and return 
+    the concatenated and unmodified resulting DataFrame.
+    
+    Args—
+    + group_id (str): groupid for fetching a given group report (definitions documented [here](https://www.caiso.com/documents/oasis-interfacespecification_v5_1_2clean_fall2017release.pdf))
+    
+    + trading_day (datetime): the day to fetch data for (group reports are always for a single day)
+    
+    Example—
+    >>> df = fetch_groupzip('DAM_LMP_GRP', datetime(2025, 4, 2)) 
+    """
+    
+    url = _build_groupzip_url(group_id, trading_day, local_tz=local_tz)
+    resp = requests.get(url, timeout=300)
```

Comment:
use the timeout in the Oasis module if provided (or overwritten there) - only if you're reasonably sure that it has not been overwritten do you clobber it with 300. Not that 300 isn't a sensible value, but hardcoding parameters makes reuse that much more painful.